src/DrDotnet.Profilers/profilers/merged_call_stacks_profiler.rs

```diff
@@ -0,0 +1,258 @@
+
+use crate::api::*;
+use crate::api::ffi::{FunctionID, HRESULT, ThreadID};
+use crate::macros::*;
+use crate::profilers::*;
+use crate::session::Report;
+
+const PADDING: usize = 5;
+const NB_THREAD_IDS_TO_PRINT: usize = 4;
+
+
+#[derive(Default)]
+pub struct MergedCallStacksProfiler {
+    clr_profiler_info: ClrProfilerInfo,
+    session_info: SessionInfo,
+    merged_stack: Arc<Mutex<MergedStack>>
+}
+
+#[derive(Default, Debug, Eq, PartialEq, Hash, Clone)]
+enum StackFrameType {
+    #[default]
+    Managed,
+    Native
+}
+
+#[derive(Default, Clone, Debug)]
+struct StackFrame {
+    display: Option<String>,
+    kind: StackFrameType,
+    fct_id: FunctionID,
+}
+
+#[derive(Default, Clone, Debug)]
+struct MergedStack {
+    thread_ids: Vec<ThreadID>,
+    stacks: Vec<MergedStack>,
+    frame: StackFrame
+}
+
+impl StackFrame {
+    fn format(frame: &StackFrame, clr: &ClrProfilerInfo) -> String {
+        match frame.kind {
+            StackFrameType::Native => "unmanaged".to_string(),
+            StackFrameType::Managed =>  unsafe { clr.get_full_method_name(frame.fct_id) }
+        }
+    }
+}
+
+impl MergedStack {
+    
+    pub fn push_thread_id(&mut self, thread_id: ThreadID) {
+        self.thread_ids.push(thread_id);
+    }
+    
+    pub fn add_stack(&mut self, clr: &ClrProfilerInfo, thread_id: ThreadID, stack_trace: Vec<StackFrame>, index: Option<usize>) {
+        self.thread_ids.push(thread_id);
+        
+        let mut index = index.unwrap_or(0);
+
+        let mut first_frame: Option<&StackFrame> = None;
+        
+        for frame in &stack_trace[index..] {
+            if frame.kind == StackFrameType::Managed { 
+                first_frame = Some(frame);
+                break
+            }
+            index += 1;
+        }
+        
+        if first_frame.is_none() {
+            return; 
+        }
+        let first_frame = first_frame.unwrap();
+        
+        let mut merged_stack = self.stacks.iter_mut()
+            .find_or_first(|s| s.frame.fct_id == first_frame.fct_id);
+        
+        if merged_stack.is_none() {
+            self.stacks.push(MergedStack::new(&first_frame, clr));
+            merged_stack = self.stacks.last_mut();
+        }
+        
+        let merged_stack = merged_stack.unwrap();
+        if index == stack_trace.len() -1 {
+            merged_stack.push_thread_id(thread_id);
+        } else {
+            merged_stack.add_stack(clr, thread_id, stack_trace, Some(index + 1));
+        }
+    }
+    
+    pub fn new(frame: &StackFrame, clr: &ClrProfilerInfo) -> Self {
+        MergedStack {
+            thread_ids: Vec::new(),
+            stacks: Vec::new(),
+            frame: StackFrame {
+                kind: frame.kind.clone(),
+                fct_id: frame.fct_id,
+                display: Some(StackFrame::format(frame, clr))
+            }
+        }
+    }
+
+    pub fn write_to(&self, report: &mut Report) {
+        self.write_stack(report, 0);
```

Comment:
Another question, do we really have 19 threads with 1 root here ? It seems there is a single thread with `System.Threading.ManualResetEventSlim.Wait` root (or maybe I don't know how to read this)

Reply:
Visualize a stack (last in, first out). Start from the bottom to read it. 
In the current output we can see that we have 19 threads and they all starts with the same root:
```
   19 JetBrains.ReSharper.TestRunner.EntryPoint.Run
   19 JetBrains.ReSharper.TestRunner.Program.Run
   19 JetBrains.ReSharper.TestRunner.Program.Main
```
5 of them have their callstack stopped at `19 JetBrains.ReSharper.TestRunner.EntryPoint.Run` and 14 threads have their callstack continue with:
```
        14 JetBrains.ReSharper.TestRunner.Bootstrapper`1.Run
        14 CommandLine.ParserResultExtensions.MapResult
```
The top line can be read as "of the 19 threads only 1 has a callstack that goes up to:
`1 System.Threading.ManualResetEventSlim.Wait` "

Reply:
Could it be that the others are in an unmanaged path after?